quickannotator/api/v1/tile/__init__.py

```diff
@@ -133,6 +140,36 @@ def get(self, args):
             tiles.extend(placeholder_tiles)
         return tiles, 200
     
+@bp.route('/search/polygon')
+class TileSearchByPolygon(MethodView):
+    @bp.arguments(SearchTileByPolygonArgsSchema, location='json')
+    @bp.response(200, TileRespSchema(many=True))
+    def post(self, args):
+        """     get all Tiles within a polygon
+        """
+        tiles_in_polygon, _, _ = get_tile_ids_intersecting_polygons(args['image_id'], args['annotation_class_id'], [args['polygon']], mask_dilation=1)
+        tile_ids_in_mask, _, _ = get_tile_ids_intersecting_mask(args['image_id'], args['annotation_class_id'], mask_dilation=1)
```

Comment:
what is the difference between this polygon and mask? isn't one of tehse supposed to be the issue mask, but that would mean the second parameter isn't set correctly?

Reply:
This is set correctly. The `get_tile_ids_intersecting_mask` function knows the mask annotation class id implicitly (it's a global variable set to 1). The function needs to know the current annotation class id so that it can get the full set of tiles for the annotation class and perform the intersection with the mask.

I can add a docstring to the function to make this more clear